### PR TITLE
Speed up unit tests a bit for operator tests

### PR DESCRIPTION
**Pull request by howardjohn**
See https://github.com/istio/istio/issues/26330

### CODE REVIEW

**Review by elfinhe — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

File: security/pkg/stsservice/tokenmanager/google/tokenexchangeplugin.go [hunk 1]
```diff
@@ -229,12 +229,14 @@ func (p *Plugin) constructFederatedTokenRequest(parameters stsservice.StsRequest
 func (p *Plugin) fetchFederatedToken(parameters stsservice.StsRequestParameters) (*federatedTokenResponse, error) {
 	respData := &federatedTokenResponse{}
 
-	req, err := p.constructFederatedTokenRequest(parameters)
-	if err != nil {
-		pluginLog.Errorf("failed to create get federated token request: %+v", err)
-		return nil, err
-	}
-	resp, timeElapsed, err := p.sendRequestWithRetry(req)
+	resp, timeElapsed, err := p.sendRequestWithRetry(func() (*http.Request, error) {
```
Comment (howardjohn) [inline security/pkg/stsservice/tokenmanager/google/tokenexchangeplugin.go:232]: Cannot reused a request, see stackoverflow link below

File: security/pkg/stsservice/test/setup.go [hunk 2]
```diff
@@ -242,6 +245,7 @@ func setupSTS(stsPort int, backendURL string, enableCache bool) (*stsServer.Serv
 	// Create token exchange Google plugin
 	tokenExchangePlugin, _ := google.CreateTokenManagerPlugin(nil, tokenBackend.FakeTrustDomain,
 		tokenBackend.FakeProjectNum, tokenBackend.FakeGKEClusterURL, enableCache)
+	tokenExchangePlugin.HTTPClient.Timeout = time.Millisecond * 5
```
Comment (howardjohn) [inline security/pkg/stsservice/test/setup.go:248]: set low timeout so tests don't take 15s each

File: security/pkg/stsservice/test/setup.go [hunk 3]
```diff
@@ -200,21 +200,24 @@ func (e *Env) WaitForStsFlowReady(t *testing.T) {
...
 				addr = stsServerAddress
 				return net.Dial(network, addr)
 			},
 		},
 	}
 	// keep sending requests periodically until a success STS response is received
-	req := e.genStsReq(stsServerAddress)
-	for i := 0; i < 20; i++ {
+	for i := 0; i < 100; i++ {
+		req := e.genStsReq(stsServerAddress)
```
Comment (howardjohn) [inline security/pkg/stsservice/test/setup.go:210]: need to make new request every time

File: pkg/test/env/ports.go [hunk 4]
```diff
@@ -85,63 +69,27 @@ type Ports struct {
...
 // NewEnvoyPorts allocate ports for Envoy
-func NewEnvoyPorts(ports *Ports, name uint16) *Ports {
-	base := allocEnvoyPortBase(name)
+func NewEnvoyPorts(t *testing.T, ports *Ports) *Ports {
 	return &Ports{
-		ClientProxyPort: base,
-		ServerProxyPort: base + 1,
-		TCPProxyPort:    base + 2,
-		AdminPort:       base + 3,
+		ClientProxyPort: portManager.ReservePortNumberOrFail(t),
```
Comment (howardjohn) [inline pkg/test/env/ports.go:89]: old method makes it very easy to run into port conflicts. New method reuse common code and removes port conflicts. This lets us do stress testing of these tests - I ran 256 concurrently

File: pkg/envoy/options.go [hunk 5]
```diff
@@ -401,7 +401,7 @@ func (bid BaseID) Close() error {
 	if bid != InvalidBaseID {
 		// Envoy internally multiplies the base ID from the command line by 10 so that they have spread
 		// for domain sockets.
-		path := fmt.Sprintf("/dev/shm/envoy_shared_memory_%d", bid.GetInternalEnvoyValue())
+		path := fmt.Sprintf("/dev/shm/envoy_shared_memory_%d", bid.GetInternalEnvoyValue()*10)
```
Comment (howardjohn) [inline pkg/envoy/options.go:404]: old code was wrong, always failed

**Review by JimmyCYJ — APPROVED** [review]

**Pull request merged**